packages/amplify-cli-core/src/types.ts

```diff
@@ -20,19 +20,209 @@ export type $TSContext = {
+  categories: Record<string, unknown>;
+};
+export type TeamProviderInfo = {
+  [envName: string]: Record<string, unknown>;
+};
+export type LocalAwsInfo = {
+  NONE: unknown;
+};
+export type ProjectConfig<T extends string = ''> = {
+  frontend: T;
+  version: string;
+  providers: string[];
+  projectPath?: string;
+  defaultEditor?: string;
+  frontendHandler?: unknown;
+} & Record<T, string>;
+export interface LocalEnvInfo {
+  projectPath: string;
+  defaultEditor: string;
+  envName: string;
+}
+export interface IFlowData {
+  setIsHeadless: (headless: boolean) => void;
+  pushHeadlessFlow: (headlessFlowDataString: string, input: $CommandLineInput) => void;
+  pushInteractiveFlow: (prompt: string, input: unknown) => void;
+  getFlowReport: () => IFlowReport | Record<string, never>;
+  assignProjectIdentifier: (envName?: string) => string | undefined;
+}
+export interface IUsageData extends IUsageMetricsData, IFlowData {}
+export type ProjectSettings = {
+  frontend?: string;
+  editor?: string;
+  framework?: string;
+};
+export interface IUsageDataPayload {
+  sessionUuid: string;
+  installationUuid: string;
+  amplifyCliVersion: string;
+  input: $CommandLineInput | null;
+  inputOptions: InputOptions;
+  timestamp: string;
+  error: SerializableError;
+  downstreamException: SerializableError;
+  payloadVersion: string;
+  osPlatform: string;
+  osRelease: string;
+  nodeVersion: string;
+  state: string;
+  isCi: boolean;
+  accountId: string;
+  projectSetting: ProjectSettings;
+  codePathDurations: Partial<Record<TimedCodePath, number>>;
+  flowReport: IFlowReport;
+  pushNormalizationFactor: number;
+}
+
+type Trace = {
+  methodName: string;
+  file: string;
+  lineNumber: string;
+  columnNumber: string;
+};
+export type SerializableError = {
+  name: string;
+  message: string;
+  details?: string;
+  code?: string;
+  trace?: Trace[];
+};
+export type InputOptions = Record<string, string | boolean>;
+export interface IFlowReport {
+  version: string;
+  runtime: string;
+  executable: string;
+  category: string;
+  isHeadless: boolean;
+  cmd: string;
+  subCmd: string | undefined;
+  optionFlowData: Array<TypeOptionFlowData>; //IOptionFlowHeadlessData | IOptionFlowCLIData
+  input: $CommandLineInput;
+  timestamp: string;
+  projectEnvIdentifier?: string; // hash(ProjectName + Amplify AppId + EnvName)
+  projectIdentifier?: string; // hash( ProjectName + Amplify App Id)
+}
+export type TimedCodePath = ManuallyTimedCodePath | UntilExitTimedCodePath | FromStartupTimedCodePaths;
+export interface IUsageMetricsData {
+  emitAbort: () => Promise<void>;
+  emitError: (error: Error | null) => Promise<void>;
+  emitSuccess: () => Promise<void>;
+  init: (
+    installationUuid: string,
+    version: string,
+    input: $CommandLineInput,
+    accountId: string,
+    projectSettings: ProjectSettings,
+    processStartTimeStamp: number,
+  ) => void;
+  getUsageDataPayload: (error: Error | null, state: string) => IUsageDataPayload;
```

Comment:
```suggestion
  getUsageData: (error: Error | null, state: string) => IUsageData;
```

or perhaps `UsageMetrics`